### DISCUSSION TITLE
How to count number of activated parameters?

### DISCUSSION THREAD

**Discussion by daeunni** [huggingface/diffusers#7036] [General]
This is just a simple question. (I am new to here 🥲) 
I am using `train_dreambooth_lora.py`, how can I count the number of activated parameters?

**Comment by a-r-r-o-w** [comment]
I haven't tested but to count number of trainable parameters for which gradients are activated, we can do something like:

```py
num_trainable = sum([param.numel() for param in model.parameters() if param.requires_grad])
```

You can add this after the weights of vae, text_encoder and unet are frozen (see [this](https://github.com/huggingface/diffusers/blob/1b3cfb1b1076f509af3592dc9d8bd76ddb41f71c/examples/dreambooth/train_dreambooth_lora.py#L809)).

**Reply by daeunni** [reply]
Thank you so much. Really helpful!
